deploying-a-django-app-on-okteto.md

```diff
@@ -0,0 +1,704 @@
+    class OnlyOwnerUser(GenericViewSet):
+        permission_classes = [IsAuthenticated, OnlyOwnerUserPermission]
+        def get(self, request):
+            user = User.objects.get(username=request.user.username)
+            user_serializer = UserSerializer(user)
+            message = "You can only see this if you're an owner"
+    
+            return Response(
+                {"user": user_serializer.data, "message": message, "success": True},
+                status=status.HTTP_200_OK,
+            )
+
+`OnlyStaffOwnerUser` can only be accessible to authenticated staff and Owner, using the permission classes `IsAuthenticated` and `OnlyStaffOwnerUserPermission`.
+
+`OnlyAdminStaffOwnerUser` is only available to admin staff and Owner. 
+`OnlyInvestorAndOwnerUser` is also only available to Investors and Owner
+`OnlyOwnerUser` is only available to owners.
+These are through the permission classes attached to them. 
+
+Add the following to your `urls.py`:
+Here were are assigning the views we created to routes:
+
+
+    from users.views import *
+    
+    urlpatterns = [
+        path('admin/', admin.site.urls),
+    ]
+    
+    urlpatterns += [
+        path('user-auth/', LoginView.as_view({"post": "login"}), name = "user-login"),
+        path('user-signup/', SignUpView.as_view({"post": "create"}), name = "user-signup"),
+        path('all-user/', AllUser.as_view({"post": "get"}), name = "all-users"),
+        path('authenticated-user/', OnlyAuthenticatedUser.as_view({"post": "get"}), name = "authenticated-user"),
+        path('only-staff/', OnlyStaffOwnerUser.as_view({"post": "get"}), name = "only-staff"),
+        path('only-admin/', OnlyAdminStaffOwnerUser.as_view({"post": "get"}), name = "only-admin"),
+        path('only-investor-owner/', OnlyInvestorAndOwnerUser.as_view({"post": "get"}), name = "only-investor-owner"),
+        path('only-owner/', OnlyOwnerUser.as_view({"post": "get"}), name = "only-owner"),
+    ]
+## Deploying to Okteto
+
+Now we will  be deploying our API and to Docker and Okteto Cloud.
+
+Get an account on Okteto and run this command;
+
+    $ okteto namespace
+
+Next we try building and deploying to Okteto:
+
+
+    $ okteto stack deploy --build
+
+This creates our Docker image from our `Dockerfile`, and builds the services in the `okteto-stack.yml` file. 
+
+Next, go to the Okteto Dashboard settings, and the secrets tab and add your secret key(gotten from your django settings or .env) for your Docker image:
+
+![](https://paper-attachments.dropbox.com/s_63C04585E62AD95AAE546582E54FC11FBBFCB4DF4EB5E31E04B3F30729CE7718_1616271618177_okteto-dashboard.png)
+
+
+Now to run migrations and set up your app finally:
+
+
+    $ okteto up
+
+Select the yes (y) option to create a `okteto.yml`. 
+
+This sets up a development container for you.  `exit` and edit line 4:
+
+    image: okteto.dev/simple-rbas:latest
+
+Now run `okteto up` again
+
+Run:
+
+    $ python manage.py makemigrations
+    $ python manage.py migrate
```

Comment:
I think it's better to put this on a script, and run it as part of the docker file, so that, when you run "okteto stack deploy --build", you have a functional application without any extra steps.

Reply:
https://github.com/okteto/math/blob/master/run_web.sh is an example of how to do that.